jina/peapods/runtimes/jinad/client.py

```diff
@@ -245,6 +139,23 @@ def _mask_args(self, args: 'argparse.Namespace'):
         # TODO:/NOTE this prevents jumping from remote to another remote (Han: 2021.1.17)
         _args.host = __default_host__
 
+        # NOTE: on remote relative filepaths should be converted to filename only
+        def basename(field):
+            if field and not field.startswith('docker://'):
+                try:
+                    return os.path.basename(complete_path(field))
+                except FileNotFoundError:
+                    pass
+            return field
+
+        for f in ('uses', 'uses_after', 'uses_before', 'uses_internal', 'py_modules'):
+            attr = getattr(_args, f, None)
+            if not attr:
+                continue
+            setattr(_args, f, [basename(m) for m in attr]) if isinstance(
+                attr, list
+            ) else setattr(_args, f, basename(attr))
+
```

Comment:
i think this part of the code especially the `basename` function has to be aligned with Hubble, in order to use Hubble images. Currently hubble's schemes are `jinahub://` or `jinahub+docker://`

Reply:
@numb3r3